python/wpt/ohos_webdriver_test.py

```diff
@@ -0,0 +1,527 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#!/usr/bin/env python3
+
+import argparse
+import json
+import logging
+import os
+import subprocess
+import time
+import sys
+import urllib.request
+import urllib.error
+from typing import Dict, Optional, Any
+
+
+class OHOSWebDriverController:
+    """Controller for running WebDriver tests on OHOS devices using HTTP API."""
+
+    def __init__(self, webdriver_port: int = 7000, wpt_server_port: int = 8000) -> None:
+        self.webdriver_port = webdriver_port
+        self.wpt_server_port = wpt_server_port
+        self.session_id: Optional[str] = None
+        self.hdc_process: Optional[subprocess.Popen] = None
+        self.wpt_server_process: Optional[subprocess.Popen] = None
+        self.wpt_reverse_port_process: Optional[subprocess.Popen] = None
+
+    def setup_wpt_server_access(self) -> bool:
+        """Set up access to WPT server for OHOS device."""
+        try:
+            cmd = ["hdc", "rport", f"tcp:{self.wpt_server_port}", f"tcp:{self.wpt_server_port}"]
+            logging.info(f"Setting up HDC reverse port forwarding for WPT: {' '.join(cmd)}")
+            self.wpt_reverse_port_process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+            time.sleep(2)
+
+            if self.wpt_reverse_port_process.poll() is not None:
+                stdout, stderr = self.wpt_reverse_port_process.communicate()
+                if self.wpt_reverse_port_process.returncode != 0:
+                    logging.warning(f"HDC reverse port forwarding failed: {stderr.decode()}")
+                    self.wpt_reverse_port_process = None
+                    return False
+
+            logging.info(f"HDC reverse port forwarding established for WPT server on port {self.wpt_server_port}")
+            return True
+
+        except FileNotFoundError:
+            logging.error("HDC command not found. Please install HDC and add it to PATH")
+            return False
+        except Exception as e:
+            logging.error(f"Failed to set up WPT server access: {e}")
+            return False
+
+    def setup_hdc_forwarding(self) -> bool:
+        """Set up HDC port forwarding for WebDriver communication."""
+        try:
+            cmd = ["hdc", "fport", f"tcp:{self.webdriver_port}", f"tcp:{self.webdriver_port}"]
+            logging.info(f"Setting up HDC port forwarding: {' '.join(cmd)}")
+            self.hdc_process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            time.sleep(2)
```

Comment:
I think it would be easier to just use `subprocess.run()` or `subprocess.check_output()`